### PR TITLE
Ensure that hash_append will never throw

### PR DESCRIPTION
**Pull request by nbougalis**
Reviewers: @HowardHinnant, @ximinez

### CODE REVIEW

**Comment by rec** [comment]
Then you should mark those accessors `noexcept` - that's the correct way to quiet the static analysis tool.

**Comment by nbougalis** [comment]
@rec I would mark them as `noexcept` except... well... they throw. We can, however, discuss whether throwing is the right solution.

File: src/beast/beast/net/IPAddress.h [hunk 1]
```diff
@@ -142,12 +142,12 @@ class Address
...
     void
-    hash_append(Hasher& h, Address const& addr)
+    hash_append(Hasher& h, Address const& addr) noexcept
     {
         using beast::hash_append;
         if (addr.is_v4 ())
             hash_append(h, addr.to_v4 ());
-        else
+        else if (addr.is_v6 ())
             hash_append(h, addr.to_v6 ());
```
Comment (ximinez) [inline src/beast/beast/net/IPAddress.h]: What if it's neither? When the proverbial IP v7 does come along, I'd hate for this function to potentially get missed and become a NOOP.

Comment (nbougalis) [inline src/beast/beast/net/IPAddress.h]: Touché @ximinez.

Comment (vinniefalco) [inline src/beast/beast/net/IPAddress.h]: @nbougalis You misspelled it

Comment (ximinez) [inline src/beast/beast/net/IPAddress.h]: My comment there is the voice of painful, bitter experience.

**Comment by vinniefalco** [comment]
@ximinez We don't even support IPV6 and you're worried about IPV7 and beyond? We need to stop nitpicking these pull requests to death and be realistic, there's 2 pages of backlog.

**Comment by ximinez** [comment]
I think my point is valid: I've seen way too many situations of "One of these conditions will always be true. Nothing will ever change that would cause this block to get skipped." Two years later, something gets changed, the block gets skipped, and because it was so long ago nobody remembers until something goes very badly. I would have been happy with `else assert(false);`

Now that I've said my piece, :+1:

**Comment by vinniefalco** [comment]
@ximinez Fair points

**Comment by nbougalis** [comment]
I'm happy to add `else assert (false);`

**Pull request closed without merging**